src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -0,0 +1,178 @@
+        # The caller should validate that it's a valid `.zip` file.
+        return (self.path, archive_path, "-d", extract_path)
+
+
+@dataclass(frozen=True)
+class GunzipBinary:
+    python: PythonBinary
+
+    def extract_archive_argv(self, archive_path: str, extract_path: str) -> tuple[str, ...]:
+        archive_name = os.path.basename(archive_path)
+        dest_file_name = os.path.splitext(archive_name)[0]
+        dest_path = os.path.join(extract_path, dest_file_name)
+        script = dedent(
+            f"""
+            import gzip
+            import shutil
+            with gzip.GzipFile(filename={archive_path!r}, mode="rb") as source:
+                with open({dest_path!r}, "wb") as dest:
+                    shutil.copyfileobj(source, dest)
+            """
+        )
+        return (self.python.path, "-c", script)
+
+
+class TarBinary(BinaryPath):
+    def create_archive_argv(
+        self, output_filename: str, input_files: Sequence[str], tar_format: ArchiveFormat
+    ) -> tuple[str, ...]:
+        # Note that the parent directory for the output_filename must already exist.
+        #
+        # We do not use `-a` (auto-set compression) because it does not work with older tar
+        # versions. Not all tar implementations will support these compression formats - in that
+        # case, the user will need to choose a different format.
+        compression = {ArchiveFormat.TGZ: "z", ArchiveFormat.TBZ2: "j", ArchiveFormat.TXZ: "J"}.get(
+            tar_format, ""
+        )
+        return (self.path, f"c{compression}f", output_filename, *input_files)
+
+    def extract_archive_argv(self, archive_path: str, extract_path: str) -> tuple[str, ...]:
+        # Note that the `output_dir` must already exist.
+        # The caller should validate that it's a valid `.tar` file.
+        return (self.path, "xf", archive_path, "-C", extract_path)
+
+
+# -------------------------------------------------------------------------------------------
+# Rules to find binaries
+# -------------------------------------------------------------------------------------------
+
+
+@rule(desc="Finding the `zip` binary", level=LogLevel.DEBUG)
+async def find_zip() -> ZipBinary:
+    request = BinaryPathRequest(
+        binary_name="zip", search_path=SEARCH_PATHS, test=BinaryPathTest(args=["-v"])
+    )
+    paths = await Get(BinaryPaths, BinaryPathRequest, request)
+    first_path = paths.first_path_or_raise(request, rationale="create `.zip` archives")
+    return ZipBinary(first_path.path, first_path.fingerprint)
+
+
+@rule(desc="Finding the `unzip` binary", level=LogLevel.DEBUG)
+async def find_unzip() -> UnzipBinary:
+    request = BinaryPathRequest(
+        binary_name="unzip", search_path=SEARCH_PATHS, test=BinaryPathTest(args=["-v"])
+    )
+    paths = await Get(BinaryPaths, BinaryPathRequest, request)
+    first_path = paths.first_path_or_raise(
+        request, rationale="download the tools Pants needs to run"
```

Comment:
Isn't the rationale slightly off here?

Being a util rule, it might not always be the case that it will be used during tool download, but simply extracting some other archive..

Reply:
Yeah it could be, you're right. Originally it was only used by `archive.py` but now we've opened the floodgates and it could be used for anything.

Not sure what a better description would be though.